src/main/kotlin/no/skatteetaten/aurora/boober/feature/S3Feature.kt

```diff
@@ -58,24 +57,27 @@ class S3Feature(
     val herkimerService: HerkimerService,
     @Value("\${application.deployment.id}") val booberApplicationdeploymentId: String,
     @Value("\${openshift.cluster}") val cluster: String,
-    @Value("\${boober.productionlevel}") productionLevel: String
-) : S3FeatureTemplate(productionLevel) {
+    @Value("\${minio.bucket.region:us-east-1}") val defaultBucketRegion: String
+) : S3FeatureTemplate() {
 
     override fun enable(header: AuroraDeploymentSpec): Boolean {
         return !header.isJob
     }
 
     override fun modify(adc: AuroraDeploymentSpec, resources: Set<AuroraResource>, cmd: AuroraContextCommand) {
-        val buckets = findS3Buckets(adc, cmd.applicationFiles).associateBy { it.name }
+        val bucketNameAndObjectAreas = findS3Buckets(adc, cmd.applicationFiles).groupBy { it.bucketName }
         resources.filter { it.resource.metadata.name.endsWith("-s3") }
             .filter { it.resource is Secret }
             .map { it.resource as Secret }
-            .forEach {
-                val bucketName = String(Base64.decodeBase64(it.data["bucketName"]))
-                val bucketNameSuffix = buckets[bucketName]?.suffix?.toUpperCase()
-                    ?: throw RuntimeException("Failed during creation of envVars, a secret with name=${it.metadata.name} has been created but the bucket does not exist")
-                val envVars = it.createEnvVarRefs(prefix = "S3_${bucketNameSuffix}_")
-                resources.addEnvVar(envVars, javaClass)
+            .forEach { secret ->
+                val bucketName = String(Base64.decodeBase64(secret.data["bucketName"]))
+
+                bucketNameAndObjectAreas[bucketName]?.map { it.name }
+                    ?.forEach { bucketObjectArea ->
+                        val envVars = secret.createEnvVarRefs(prefix = "S3_${bucketObjectArea}_")
+                        resources.addEnvVar(envVars, javaClass)
+                    }
+                    ?: throw RuntimeException("Failed during creation of envVars, a secret with name=${secret.metadata.name} has been created but the bucket or objectArea does not exist")
```

Comment:
Burde ikke denne feilen komme under validation?

Reply:
Dette skal egentlig ikke skje, siden det ikke er mulig at det er skrevet en secret for en S3BucketObjectArea som ikke er konfigurert i AuroraConfig.Mulig vi bør gjøre !!, men er jo ikke så bra det heller.  Kan ikke gjøre dette i validate, siden vi er avhengig av secrets, dvs da må vi egentlig gjøre hele generate i validation.